pkg/virt-controller/services/template_test.go

```diff
@@ -4458,6 +4462,222 @@ var _ = Describe("Template", func() {
+								TypeMeta: metav1.TypeMeta{
+									Kind: "Namespace",
+								},
+								ObjectMeta: metav1.ObjectMeta{
+									Name: "custom-memory-ratio-ns",
+									Labels: map[string]string{
+										v1.AutoMemoryLimitsRatioLabel: ratioLabelValue,
+									},
+								},
+							}
+							err := namespaceStore.Add(&namespaceWithCustomMemoryRatio)
+							Expect(err).ToNot(HaveOccurred())
+
+							sampleQuantity := resource.MustParse("128M")
+							resourceQuotaWithMemoryLimits := kubev1.ResourceQuota{
+								ObjectMeta: metav1.ObjectMeta{
+									Namespace: "custom-memory-ratio-ns",
+								},
+								Spec: kubev1.ResourceQuotaSpec{
+									Hard: kubev1.ResourceList{
+										kubev1.ResourceLimitsMemory: sampleQuantity,
+									},
+								},
+							}
+							err = resourceQuotaStore.Add(&resourceQuotaWithMemoryLimits)
+							Expect(err).ToNot(HaveOccurred())
+
+							vmi := v1.VirtualMachineInstance{
+								ObjectMeta: metav1.ObjectMeta{
+									Name:      "testvmi",
+									Namespace: "custom-memory-ratio-ns",
+									UID:       "1234",
+								},
+								Spec: v1.VirtualMachineInstanceSpec{
+									Domain: v1.DomainSpec{
+										Resources: v1.ResourceRequirements{
+											Requests: kubev1.ResourceList{kubev1.ResourceMemory: guestMemory},
+										},
+									},
+								},
+							}
+
+							pod, err := svc.RenderLaunchManifest(&vmi)
+							Expect(err).ToNot(HaveOccurred())
+							Expect(pod.Spec.Containers[0].Name).To(Equal("compute"))
+							expectedValue := int64(float64(pod.Spec.Containers[0].Resources.Requests.Memory().Value()) * expectedUsedRatio)
+							Expect(pod.Spec.Containers[0].Resources.Limits.Memory().Value()).To(BeEquivalentTo(expectedValue))
+						},
+							Entry("using default ratio value if the label value is not a float", "not_a_float", DefaultMemoryLimitOverheadRatio),
+							Entry("using custom ratio value if the label value is a float", "3.2", 3.2),
+						)
+					})
+				})
+			})
+
+			Context("when the creation namespace have a resource quota without memory limits associated to it", func() {
+				It("should not set memory limits", func() {
+					sampleQuantity := resource.MustParse("128M")
+					resourceQuotaWithoutMemoryLimits := kubev1.ResourceQuota{
+						ObjectMeta: metav1.ObjectMeta{
+							Namespace: "no-rq-namespace",
+						},
+						Spec: kubev1.ResourceQuotaSpec{
+							Hard: kubev1.ResourceList{
+								kubev1.ResourceMemory: sampleQuantity,
+							},
+						},
+					}
+					err := resourceQuotaStore.Add(&resourceQuotaWithoutMemoryLimits)
+					Expect(err).ToNot(HaveOccurred())
+
+					vmi := v1.VirtualMachineInstance{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      "testvmi",
+							Namespace: "no-rq-namespace",
+							UID:       "1234",
+						},
+						Spec: v1.VirtualMachineInstanceSpec{
+							Domain: v1.DomainSpec{
+								Memory: &v1.Memory{
+									Guest: &guestMemory,
```

Comment:
Can we also have a positive test that uses `Spec.Domain.Memory.Guest` instead of `Spec.Domain.Resources.Requests`?

Reply:
Well, actually this show something wrong in the test.
Basically, at virt-controller level, we only care about the  `Spec.Domain.Resources.Requests`.
This because the logic between `Spec.Domain.Memory.Guest` vs `Spec.Domain.Resources.Requests` is in the mutating webhook https://github.com/kubevirt/kubevirt/blob/main/pkg/virt-api/webhooks/defaults.go#L125-L178; basically, if we use `Guest` or `Hugepages.PageSize` and not `Requests`, we set the latter with a proper value. This means that  `Spec.Domain.Resources.Requests` is always valued at this level of computation and `Guest` doesn't matter anymore.

As I said, this shows that here I should use `Spec.Domain.Resources.Requests` instead of `Spec.Domain.Memory.Guest`.
Do you agree? Thank you!

Reply:
All good, still found an issue :D